### PR TITLE
refactor `blas/ext/base/dcusumkbn` to follow current project conventions

### LINKED ISSUE #1472: [RFC]: refactor `blas/ext/base/dcusumkbn` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/dcusumkbn`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/dcusumkbn) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by itskdhere**
Resolves #1472 .

## Description

> What is the purpose of this pull request?

This pull request:
-   refactors the existing implementation of `blas/ext/base/dcusumkbn` to follow the current project convention.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1472

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-  [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/README.md [hunk 1]
```diff
@@ -65,18 +65,15 @@ The function has the following parameters:
 -   **y**: output [`Float64Array`][@stdlib/array/float64].
 -   **strideY**: index increment for `y`.
 
-The `N` and `stride` parameters determine which elements in `x` and `y` are accessed at runtime. For example, to compute the cumulative sum of every other element in `x`,
+The `N` and `stride` parameters determine which elements in the strided arrays are accessed at runtime. For example, to compute the cumulative sum of every other element in the strided input array,
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/README.md]:
```suggestion
The `N` and stride parameters determine which elements in the strided arrays are accessed at runtime. For example, to compute the cumulative sum of every other element in the strided input array,
```

File: lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/src/dcusumkbn.c [hunk 2]
```diff
@@ -38,7 +38,7 @@
 * @param Y        output array
 * @param strideY  Y stride length
 */
-void stdlib_strided_dcusumkbn( const int64_t N, const double sum, const double *X, const int64_t strideX, double *Y, const int64_t strideY ) {
+double stdlib_strided_dcusumkbn( const int64_t N, const double sum, const double *X, const int64_t strideX, double *Y, const int64_t strideY ) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/src/dcusumkbn.c:41]: This change in return type should be reverted.

File: lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/include/stdlib/blas/ext/base/dcusumkbn.h [hunk 3]
```diff
@@ -34,7 +34,7 @@ extern "C" {
 /**
 * Computes the cumulative sum of double-precision floating-point strided array elements using an improved Kahan–Babuška algorithm.
 */
-void stdlib_strided_dcusumkbn( const int64_t N, const double sum, const double *X, const int64_t strideX, double *Y, const int64_t strideY );
+double stdlib_strided_dcusumkbn( const int64_t N, const double sum, const double *X, const int64_t strideX, double *Y, const int64_t strideY );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/include/stdlib/blas/ext/base/dcusumkbn.h:37]: This change in return type should be reverted.

Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/include/stdlib/blas/ext/base/dcusumkbn.h]: This change in return type should be reverted.

```suggestion
void stdlib_strided_dcusumkbn( const int64_t N, const double sum, const double *X, const int64_t strideX, double *Y, const int64_t strideY );
```

File: lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/manifest.json [hunk 4]
```diff
@@ -1,5 +1,7 @@
 {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/manifest.json:1]: Need two-space indentation here.

File: lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/src/dcusumkbn.c (hunk 2 above)
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/src/dcusumkbn.c]: This change in return type should be reverted.

```suggestion
void stdlib_strided_dcusumkbn( const int64_t N, const double sum, const double *X, const int64_t strideX, double *Y, const int64_t strideY ) {
```

File: lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/test/test.dcusumkbn.js [hunk 5]
```diff
@@ -36,7 +35,7 @@ tape( 'main export is a function', function test( t ) {
 });
 
 tape( 'the function has an arity of 6', function test( t ) {
-	t.strictEqual( dcusumkbn.length, 6, 'has expected arity' );
+	t.strictEqual( dcusum.length, 6, 'returns expected value' );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/test/test.dcusumkbn.js]: Copy-paste error:

```suggestion
	t.strictEqual( dcusumkbn.length, 6, 'returns expected value' );
```

File: lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/test/test.dcusumkbn.native.js [hunk 6]
```diff
@@ -45,7 +44,7 @@ tape( 'main export is a function', opts, function test( t ) {
 });
 
 tape( 'the function has an arity of 6', opts, function test( t ) {
-	t.strictEqual( dcusumkbn.length, 6, 'has expected arity' );
+	t.strictEqual( dcusum.length, 6, 'returns expected value' );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/test/test.dcusumkbn.native.js]: Copy-paste error:

```suggestion
	t.strictEqual( dcusumkbn.length, 6, 'returns expected value' );
```

File: lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/test/test.ndarray.js [hunk 7]
```diff
@@ -36,7 +35,7 @@ tape( 'main export is a function', function test( t ) {
 });
 
 tape( 'the function has an arity of 8', function test( t ) {
-	t.strictEqual( dcusumkbn.length, 8, 'has expected arity' );
+	t.strictEqual( dcusum.length, 8, 'returns expected value' );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/test/test.ndarray.js]:
```suggestion
	t.strictEqual( dcusumkbn.length, 8, 'returns expected value' );
```

File: lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/test/test.ndarray.native.js [hunk 8]
```diff
@@ -45,7 +44,7 @@ tape( 'main export is a function', opts, function test( t ) {
 });
 
 tape( 'the function has an arity of 8', opts, function test( t ) {
-	t.strictEqual( dcusumkbn.length, 8, 'has expected arity' );
+	t.strictEqual( dcusum.length, 8, 'returns expected value' );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/test/test.ndarray.native.js]:
```suggestion
	t.strictEqual( dcusumkbn.length, 8, 'returns expected value' );
```

**Review by Planeshifter — CHANGES_REQUESTED** [review]
Thanks for your PR! 

Did an initial review and flagged some issues that will have to be addressed before merging.

**Comment by itskdhere** [comment]
@Planeshifter I've made the requested changes.

**Comment by itskdhere** [comment]
I think, I made some big mistakes: [64008be](https://github.com/stdlib-js/stdlib/pull/2025/commits/64008bec1e5a3c1b18857360f1679b58f4ccffd3) & [64ddf87](https://github.com/stdlib-js/stdlib/pull/2025/commits/64ddf871808edcaa4f476927e9df50ec8257e9ab)

Should I open a new PR ?

**Comment by Pranavchiku** [comment]
If you are unable to revert those here, please feel free to open a new PR and close this referring to new one.

**Comment by itskdhere** [comment]
> If you are unable to revert those here, please feel free to open a new PR and close this referring to new one.

Ok, working on it. Thanks for the clarification.

**Comment by itskdhere** [comment]
I'm closing this PR and opened a new PR #2036 . Please take a look into that:
- #2036

<br>

And sorry for the inconvenience 🙏

**Pull request closed without merging**